pkg/registry/apis/secret/decrypt/grpc_client.go

```diff
@@ -154,7 +135,7 @@ func (g *GRPCDecryptClient) Decrypt(ctx context.Context, namespace string, names
 	// Decryption will still use the service identity from the auth token,
 	// but we also pass the service identity from the request metadata for auditing purposes.
 	md := metadata.New(map[string]string{
-		contracts.HeaderGrafanaServiceIdentityName: serviceIdentity,
+		contracts.HeaderGrafanaServiceIdentityName: serviceName,
```

Comment:
Anyway the actual service identity that's used will be from the auth token, so we dont need to check for any explicit identity in the client side here.